File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def dfs(result, depth):
    global answer, n, t
    if depth == len(n) and result == t:
        answer += 1
    elif depth < len(n):
        for i in range(1, 3):
            result += n[depth]*((-1)**i)
            dfs(result, depth+1)
            result -= n[depth]*((-1)**i)
            
q = []
def solution(numbers, target):
    global answer, n, t
    n = numbers
    t = target
    
    dfs(0, 0)
    return answer

answer = 0
n = []
t = 0

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;
class Solution {
    ArrayList<Integer> comb = new ArrayList<>();
    int answer = 0;
    int t;
    public void dfs(int depth, int[] numbers) {
        if (depth == numbers.length) {
            int sum = 0;
            for (int i = 0; i < comb.size(); i++)
                sum += comb.get(i);
            if (sum == t)
                answer += 1;
        } else {
            comb.add(numbers[depth]);
            dfs(depth+1, numbers);
            comb.remove(comb.size()-1);
            
            comb.add(-1 * numbers[depth]);
            dfs(depth+1, numbers);
            comb.remove(comb.size()-1);
        }
    }
    public int solution(int[] numbers, int target) {
        t = target;
        dfs(0, numbers);
        return answer;
    }
}